In [2]:
import scipy
print(scipy.__version__)

1.17.1


In [3]:
from scipy.stats import chi2_contingency

In [9]:
# Contingency table:
# [converted, not converted]
# A_basic vs B_premium pricing groups

contingency = [
    [4673, 20177],   # A_basic
    [3817, 21034]    # B_premium
]

In [6]:
print(contingency)

[[4673, 20177], [3817, 21034]]


In [7]:
# Run chi-square test to evaluate whether
# the difference in conversion rates is statistically significant

chi2, p, dof, expected = chi2_contingency(contingency)

In [8]:
# Print test results

print(f"chi-square: {chi2:.2f}")
print(f"p-value: {p:.10f}")

chi-square: 103.88
p-value: 0.0000000000


A chi-square test confirmed that the difference in conversion rates between pricing groups was statistically significant (χ² = 103.88, p < 0.001), suggesting that pricing had a measurable impact on customer behavior.